substrate/frame/staking-async/src/pallet/mod.rs

```diff
@@ -2560,5 +2684,44 @@ pub mod pallet {
 
 			Ok(Pays::No.into())
 		}
+
+		/// Perform one step of era pruning to prevent DoS from unbounded deletions.
+		///
+		/// This extrinsic enables permissionless lazy pruning of era data by performing
+		/// incremental deletion of storage items. Each call processes a limited number
+		/// of items based on available block weight to avoid DoS attacks.
+		///
+		/// Returns `Pays::No` when work is performed to incentivize regular maintenance.
+		/// Anyone can call this to help maintain the chain's storage health.
+		///
+		/// The era must be eligible for pruning (older than HistoryDepth + 1).
```

Comment:
Since calling this incorrectly will incur a fee, should probably mention how to check whether pruning for a given `era` is needed or not?